docker-image/Dockerfile.amd64

```diff
@@ -16,36 +16,142 @@
 # NOTE: This Dockerfile should be kept in-sync with the one in calico/node.
 # This ensures that testing of Felix in this repository is done in the same
 # userspace environment as it will be deployed in calico/node.
+
+ARG ARCH=x86_64
+ARG GIT_VERSION=unknown
+ARG IPTABLES_VER=1.8.2-16
+ARG RUNIT_VER=2.1.2
+ARG BIRD_IMAGE=calico/bird:latest
+
 FROM calico/bpftool:v5.3-amd64 as bpftool
 
-FROM debian:10-slim
-LABEL maintainer="Shaun Crampton <shaun@tigera.io>"
+# Use this build stage to build iptables rpm and runit binaries.
+# We need to rebuild the iptables rpm because the prepackaged rpm does not have legacy iptables binaries.
+# We need to build runit because there aren't any rpms for it in CentOS or ubi repositories.
+FROM centos:8 as centos
+
+ARG ARCH
+ARG IPTABLES_VER
+ARG RUNIT_VER
+ARG CENTOS_MIRROR_BASE_URL=http://vault.centos.org/8.1.1911
+ARG LIBNFTNL_VER=1.1.1-4
+ARG LIBNFTNL_SOURCERPM_URL=${CENTOS_MIRROR_BASE_URL}/BaseOS/Source/SPackages/libnftnl-${LIBNFTNL_VER}.el8.src.rpm
+ARG IPTABLES_SOURCERPM_URL=${CENTOS_MIRROR_BASE_URL}/BaseOS/Source/SPackages/iptables-${IPTABLES_VER}.el8.src.rpm
+
+# Install build dependencies and security updates.
+RUN dnf install -y 'dnf-command(config-manager)' && \
```

Comment:
What about moving all this complex shell out into a helper script?  Then you'd just have `RUN <my helper>`  (Embedding lots of one language in another a code smell since it requires lots of escaping and it defeats linting and syntax highlighting.)